packages/vite/src/node/plugins/define.ts

```diff
@@ -46,7 +46,8 @@ export function definePlugin(config: ResolvedConfig): Plugin {
       SSR: !!config.build.ssr
     }
     for (const key in env) {
-      importMetaKeys[`import.meta.env.${key}`] = JSON.stringify(env[key])
+      const envVal = env[key]
+      importMetaKeys[`import.meta.env.${key}`] = typeof envVal === 'string' ? `\`${envVal}\`` : JSON.stringify(envVal)
```

Comment:
I don't think this is the right fix. If the `import.meta.env.xxx` is in a string, it should be skip entirely, which is being fixed in https://github.com/vitejs/vite/pull/9791